library/std/src/sys/windows/stdio.rs

```diff
@@ -29,6 +31,32 @@ struct IncompleteUtf8 {
     len: u8,
 }
 
+impl IncompleteUtf8 {
+    // Implemented for use in Stdin::read.
+    fn read(&mut self, buf: &mut [u8]) -> usize {
+        // Write to buffer until the buffer is full or we run out of bytes.
+        let to_write = cmp::min(buf.len(), self.len as usize);
+        for (ind, val) in self.bytes[..to_write].iter().enumerate() {
+            buf[ind] = *val;
+        }
```

Comment:
This can also be simplified using [`copy_from_slice`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.copy_from_slice
):

```rust
buf[..to_write].copy_from_slice(&self.bytes[..to_write])
```